### ISSUE TITLE
Test if test url is covered by any target

### ISSUE THREAD

**Issue by J0WI** [EFForg/https-everywhere#14290]
Test urls with hosts not related to any target are currently allowed in rulesets.
The following scenario should should fail in ruleset tests:
```
<ruleset name="example">
	<target host="target.example" />

	<test url="http://no-target.example/" />

	<rule from="^http:" to="https:" />
</ruleset>
```

Type: other

**Comment by jeremyn** [comment]
This might be challenging to check completely, for example for a wildcard `target` with a very complex `rule`.

Have you seen rulesets like this in the wild?

**Comment by J0WI** [comment]
One example has been fixed here: #14177

**Comment by cschanaj** [comment]
Submitted a PR in #14293 (not really sure if I have the implementation correct.. though)

@Hainish there is something was with `chrome-beta` and `chrome-stable` since https://github.com/EFForg/https-everywhere/commit/d36ea408818d865966b8cd632b065a1f1b5a88fc. Would you mind to take a look?

**Comment by Hainish** [comment]
@cschanaj looking now

**Comment by Hainish** [comment]
@cschanaj this should be fixed with 7f4fe7c8547aa7dabb0762c268b8b67857b91182, thanks for alerting me to this.
